### PR TITLE
IDEX-3241 Add che admin war

### PR DESCRIPTION
**Pull request by benoitf**
Change-Id: I541ee0a4de92d7f140ef3b904badbe7ded0682b8
Signed-off-by: Florent BENOIT fbenoit@codenvy.com

### CODE REVIEW

**Comment by garagatyi** [comment]
Can you explaine why you used AdminContextListener instead of our default configuration of Guice?

**Comment by benoitf** [comment]
I guess you're talking about CodenvyBootstrap class ?
AFAIK I'm on Che so no Codenvy use.

Florent

**Comment by garagatyi** [comment]
But main war is using CodenvyBootstrap. We should use the same bootstraps everywhere. If we need so we should rename CodenvyBootstrap

**Comment by benoitf** [comment]
I already raised the issue last week https://jira.codenvycorp.com/browse/IDEX-3191

**Comment by garagatyi** [comment]
Then just use CodenvyBootstrap and wait until it is renamed

**Comment by benoitf** [comment]
OK if you want.
BTW this is the only issue ?

**Comment by benoitf** [comment]
By using CodenvyBootstrap (and moving getModules() to a DynaModule annotated class, JAX-RS service is no longer there. (But Plugins module is analyzed and added as Guice module)
So I suppose there is something wrong inside CodenvyBootstrap :-) but as there are no trace or errors

I would say that this is because Bootstrap it is removing the mapping on the Everrest Servlet
return new ServletModule() {
            protected void configureServlets() {
                this.serve("/*", new String[0]).with(GuiceEverrestServlet.class);
            }
        };
But I don't want to add like ugly filter

**Comment by garagatyi** [comment]
CodenvyBootstrap works in other wars, so I suppose there is something wrong with configuration or code.
BTW it is not mandatory that @Dynamodule should be used. Module can be installed manually in other module.

**Comment by benoitf** [comment]
I've added CodenvyBootstrap class but it makes more burden:
- It requires to add a new class that implements ServletModule
- It imports a Codenvy class while we're on Che

File: che-admin-war/pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-ide-parent</artifactId>
+        <groupId>org.eclipse.che.ide</groupId>
+        <version>3.13.1-SNAPSHOT</version>
+    </parent>
+    <artifactId>che-admin-war</artifactId>
```
Comment (skabashnyuk) [inline che-admin-war/pom.xml:21]: Not sure about artifact name and location. @gazarenkov wdyt?

Comment (gazarenkov) [inline che-admin-war/pom.xml:21]: We definitely need to rename CodenvyBootstrap and review all the dependencies in order to be "Che" and to use same bootstrap mechanism. It is important.

Comment (benoitf) [inline che-admin-war/pom.xml:21]: It is using same bootstrap mechanism

**Comment by skabashnyuk** [comment]
LGTM you can merge it in master

**Pull request merged**